timApp/velp/velp.py

```diff
@@ -834,6 +839,31 @@ def create_default_velp_group_route(doc_id: int) -> Response:
     return json_response(created_velp_group)
 
 
+@velps.delete("/delete_velp_group_from_database/<int:group_id>")
+def delete_velp_group_from_database(group_id: int) -> Response:
+    """Remove database entries for the specified velp group.
+
+    :param group_id: Unique id of the velp group.
+    :return: OK response, if the operation was successful.
+    """
+    d = get_doc_or_abort(group_id)
+    verify_ownership(d)
+
+    vg = VelpGroup.query.filter_by(id=group_id).first()
+    v_in_g = VelpInGroup.query.filter_by(velp_group_id=group_id).all()
+    vg_sel = VelpGroupSelection.query.filter_by(velp_group_id=group_id).all()
+    vg_def = VelpGroupDefaults.query.filter_by(velp_group_id=group_id).all()
+    vg_in_doc = VelpGroupsInDocument.query.filter_by(velp_group_id=group_id).all()
+
+    targets = [*vg_in_doc, *v_in_g, *vg_sel, *vg_def, vg]
+
+    for t in targets:
+        db.session.delete(t)
```

Comment:
Onnistuisikohan tässä ihan suoraan `DELETE FROM`-kyselyn tekeminen? SQLAlchemyssä sellaisen saa aikaan esimerkiksi

```python
VelpInGroup.query.filter_by(velp_group_id=group_id).delete(synchronize_session=False)
```

(tämä tekee `DELETE`-operaation, joka poistaa VelpInGroup-taulusta kaikki filtteröidyt rivit)

Ainoastaan siinä joutuisi ehkä katsomaan, että poistot tekee oikeassa järjestyksessä. SQLAlchemy taitaa jokaisen `DELETE`n kohdalla katsoa, että kantaan ei jää mahdollisesti epävalideja rivejä tauluihin (eli rivit, josta poiston takia jouduttaisiin poistamaan joku sarake vaikka näin olisi estetty).

Jos `delete()` saa toimimaan, olisi se siis parempi, koska se ei hae tietokannasta yhtään dataa Pythoniin (nopeuttaa poistoa). Tosiaankin, `all()` hakee kannasta kaikki rivit, mikä on poiston kannalta vähän turhaa.

Reply:
OK, minä kokeilen muuttaa noin, mulla oli mennyt tuo ohi SQLAlchemyn dokkareista.